drivers/spi/spi_silabs_eusart.c

```diff
@@ -15,33 +16,68 @@
 #include <zephyr/drivers/clock_control/clock_control_silabs.h>
 #include <zephyr/drivers/pinctrl.h>
 #include <zephyr/logging/log.h>
-
+#ifdef CONFIG_SPI_ASYNC
+#include <zephyr/drivers/dma.h>
+#if defined(CONFIG_PM_DEVICE)
+#include <zephyr/pm/policy.h>
+#endif
+#endif
 #include <em_cmu.h>
 #include <em_eusart.h>
 
 LOG_MODULE_REGISTER(spi_silabs_eusart, CONFIG_SPI_LOG_LEVEL);
 
 #include "spi_context.h"
 
+/* Macro Declarations */
 #define SPI_WORD_SIZE 8
 
+#ifdef CONFIG_SPI_ASYNC
+#define SPI_MAX_DMA_DESCRIPTOR_PER_CHANNEL   (8U)
+#define SPI_DMA_MAX_DESCRIPTOR_TRANSFER_SIZE (0x800U)
+#define SPI_TX_DUMMY_BYTE_VALUE              (0xFFU)
+#endif /* CONFIG_SPI_ASYNC */
+
 /* Structure Declarations */
 
 struct spi_silabs_eusart_data {
 	struct spi_context ctx;
+#ifdef CONFIG_SPI_ASYNC
+	uint32_t dma_rx_channel;
+	uint32_t dma_tx_channel;
+	struct dma_config dma_tx_config;
+	struct dma_block_config dma_tx_descriptors[SPI_MAX_DMA_DESCRIPTOR_PER_CHANNEL];
+	struct dma_config dma_rx_config;
+	struct dma_block_config dma_rx_descriptors[SPI_MAX_DMA_DESCRIPTOR_PER_CHANNEL];
+#endif /* CONFIG_SPI_ASYNC */
 };
 
 struct spi_silabs_eusart_config {
 	EUSART_TypeDef *base;
+#ifdef CONFIG_SPI_ASYNC
+	const struct device *dma_device;
+#endif
 	const struct device *clock_dev;
 	const struct silabs_clock_control_cmu_config clock_cfg;
 	uint32_t clock_frequency;
 	const struct pinctrl_dev_config *pcfg;
 };
 
+/* Global variable */
+#ifdef CONFIG_SPI_ASYNC
+static volatile uint8_t dummy_byte = SPI_TX_DUMMY_BYTE_VALUE;
```

Comment:
Should use `SPI_MOSI_OVERRUN_DT_INST`, which gets seeded from the `overrun-character` property in DTS.